src/components/shared/TheExercise.vue

```diff
@@ -11,27 +11,32 @@
                         <slot></slot>
                     </div>
                 </div>
-                <div class="flex gap-3 max-[400px]:flex-col max-[500px]:gap-1">
+                <div class="flex gap-2 max-[400px]:flex-col max-[500px]:gap-1">
                     <p class="text-center"><span class="font-bold">Sets: </span>{{ sets }}</p>
                     <p class="text-center"><span class="font-bold">Reps: </span>{{ reps }}</p>
-                    <p class="text-center"><span class="font-bold">Weight: </span>{{ weight }}</p>
+                    <p class="text-center"><span class="font-bold">Weight: </span>{{ weight }} {{ unit }}</p>
                 </div>
                 <p><span class="font-bold">Rest time: </span>{{ rest }}</p>
             </div>
         </div>
     </div>
 </template>
 
-<script>
-export default {
-  name: 'TheExercise',
-  props: ['img', 'name', 'sets', 'reps', 'weight', 'rest']
-}
+<script setup lang="ts">
+defineProps<{
+  name: string,
+  img: string,
+  sets: string | undefined,
+  reps: string | undefined,
+  weight: string | undefined,
+  unit: string | undefined,
+  rest: string | undefined
```

Comment:
I think you can easily change those to optional props.

```suggestion
  sets?: string,
  reps?: string,
  weight?: string,
  unit?: string,
  rest?: string
```